aopy/analysis/behavior.py

```diff
@@ -273,3 +274,67 @@ def calc_segment_duration(events, event_times, start_events, end_events, target_
     target_codes = np.array([trial_events[trial_idx][idx] for trial_idx, idx in enumerate(target_idx)]) - np.min(target_codes)
 
     return segment_duration, target_codes
+
+def calc_tracking_rewards(preproc_dir, subject, te_id, date, reward_interval=None, trial_start_code=2, reward_code=48, cursor_enter_target_code=80, cursor_leave_target_code=96):
+    '''
+    Calculates the maximum possible and actual number of tracking rewards acquired during reward trials.
+
+    Args:
+        preproc_dir (str): base directory where the files live
+        subject (str): Subject name
+        te_id (int): Block number of task entry object
+        date (str): Date of recording
+        reward_interval (float, optional): length of time (in s) between rewards while cursor is in target (i.e. while tracking in)
+        cursor_enter_target_code (int, optional): event code for cursor entering target
+        cursor_leave_target_code (int, optional): event code for cursor leaving target
+        reward_code (int, optional): event code for reward
+
+    Returns:
+        tuple: tuple containing:
+        | **tracking_rewards (ntrials):** number of tracking rewards acquired per reward trial
+        | **max_rewards (int):** maximum possible number of tracking rewards on a single trial
+    '''
+    # load preproc data file
+    exp_data, exp_metadata = postproc.load_preproc_exp_data(preproc_dir, subject, te_id, date)
+    assert 'tracking_rewards' in [feature.decode("utf-8") for feature in exp_metadata['features']], "No tracking rewards"
+    if reward_interval is None:
+        reward_interval = exp_metadata['tracking_reward_interval']
+    event_codes = exp_data['events']['code']
+    event_times = exp_data['events']['timestamp']
+
+    # calculate max possible number of tracking rewards
+    seq_params = lookup_sessions(id=te_id, subject=subject, date=date)[0].sequence_params
```

Comment:
i want to avoid putting database functions inside analysis functions so we can share these functions outside the lab if necessary. you can get the sequence params from exp_metadata too, they just need to be converted from json to dict using json.loads()